### PR TITLE
Support Scala Native

### PR DESCRIPTION
**Pull request by lolgab**
- Add Scala 2.11 Either compatibility wrapper for Scala Native
- Use libcrypto md5 implementation. It can be static or dynamic linked

Blocked by:
  - [x] https://github.com/lihaoyi/os-lib/pull/34 ~Missing version on Sonatype~
  - [x] https://github.com/scopt/scopt/pull/283
  - [x] https://github.com/lihaoyi/mill/pull/812
  - [x] https://github.com/lihaoyi/PPrint/pull/52

### CODE REVIEW

**Comment by hepin1989** [comment]
@lolgab Any progress?

**Comment by lolgab** [comment]
@hepin1989 I need to rebase it on top of last changes. I have to find the time to actually do it.

File: sjsonnet/src-native/sjsonnet/Platform.scala [hunk 1]
```diff
@@ -0,0 +1,15 @@
+package sjsonnet
+
+object Platform {
+  def gzipBytes(b: Array[Byte]): String = GZIP.gzipBytes(b)
+  def gzipString(s: String): String = {
+    gzipBytes(s.getBytes())
+  }
+  def xzBytes(b: Array[Byte]): String = throw new Exception("XZ not implemented in Scala Native")
+  def xzString(s: String): String = throw new Exception("XZ not implemented in Scala Native")
```
Comment (lolgab) [inline sjsonnet/src-native/sjsonnet/Platform.scala:9]: XZ is actually implementable in Scala Native using lzma. I will do it in a follow up PR

File: build.sc [hunk 2]
```diff
@@ -1,139 +1,211 @@
...
+    def libcrypto = T(os.walk(os.root / "usr").filter(_.last == "libcrypto.a").head.toString)
+    def staticLinkLibcrypto = T { true }
+    def libcryptoLinkingOptions = T {
+      if(staticLinkLibcrypto()) {
+        Array(libcrypto())
+      } else {
+        val libcryptoDir = os.Path(libcrypto()) / os.up
+        Array(s"-L$libcryptoDir", "-lcrypto")
+      }
+    }
```
Comment (lolgab) [inline build.sc:22]: This can be removed if we allow only dynamic linking, this means that users will need to install openssl to use sjsonnet.

**Comment by lolgab** [comment]
`./mill __.test` passes now.
@lihaoyi This is ready for review :-)

File: build.sc (hunk 2 above)
Comment (lihaoyi) [reply build.sc:22]: @lolgab maybe let's just stub out the MD5 implementation with `???` for now, I think that's what we're already doing on the JS side of things

**Comment by lolgab** [comment]
Superseded by #109

**Pull request closed without merging**